logspec-worker/logspec_api.py

```diff
@@ -204,7 +218,25 @@ def process_log(log_file, parser, start_state):
     return get_logspec_errors(parsed_data, parser)
 
 
-def generate_issues_and_incidents(result_id, log_file, test_type, origin):
+def look_up_existing_issue(cursor, origin, signature_loc):
+    try:
+        query = (
+            "SELECT id, version FROM issues"
+            " WHERE misc->'logspec'->'error'->>'signature_loc' = %s"
+            " AND origin = %s"
+            " AND _timestamp >= NOW() - INTERVAL '5 days'"
+        )
+        cursor.execute(query, (signature_loc, origin))
+        issues = cursor.fetchall()
+        if issues:
+            logging.debug("Issues: %r", issues)
+            return issues[0]
+    except Exception as e:
+        logging.exception(f"Error fetching unprocessed builds: {e}")
```

Comment:
nit: logging has lazy formatting

```suggestion
        logging.exception("Error fetching unprocessed builds: %s", e)
```